packages/components/psammead-grid/src/index.jsx

```diff
@@ -0,0 +1,219 @@
+    `;
+  }
+  if (min) {
+    return `
+      @media (min-width: ${min}) {
+        ${styles}
+      }
+    `;
+  }
+  if (max) {
+    return `
+      @media (max-width: ${max}) {
+        ${styles}
+      }
+    `;
+  }
+  return '';
+};
+
+const gelMaxWidths = css`
+  margin: 0 auto;
+  @media (min-width: ${GEL_GROUP_4_SCREEN_WIDTH_MIN}) and (max-width: ${GEL_GROUP_4_SCREEN_WIDTH_MAX}) {
+    max-width: ${GROUP_4_MAX_WIDTH};
+  }
+  @media (min-width: ${GEL_GROUP_5_SCREEN_WIDTH_MIN}) {
+    max-width: ${GROUP_5_MAX_WIDTH};
+  }
+`;
+
+const group4MaxWidth = css`
+  margin: 0 auto;
+  @media (min-width: ${GEL_GROUP_4_SCREEN_WIDTH_MIN}) {
+    max-width: ${GROUP_4_MAX_WIDTH};
+  }
+`;
+
+const gridMediaQueries = ({
+  columns,
+  startOffset,
+  enableGelGutters,
+  enableGelMargins,
+}) => {
+  const selectedGroups = Object.keys(columns);
+  return selectedGroups.map(group =>
+    mediaQuery({
+      min: groups[group].min,
+      max: groups[group].max,
+      styles: `
+        width: initial;
+        grid-template-columns: repeat(${columns[group]}, 1fr);
+        grid-column-end: span ${columns[group]};
+      ${enableGelGutters ? `grid-column-gap: ${groups[group].gutterSize};` : ``}
+      ${enableGelMargins ? `padding: 0 ${groups[group].marginSize};` : ``}
+      ${
+        startOffset && startOffset[group]
+          ? `grid-column-start: ${startOffset[group]};`
+          : ``
+      }`,
+    }),
+  );
+};
+
+/*
+ * 1 We vertically align to the top so that sibling
+ *   grid items that are placed side-by-side within a row
+ *   have their text and images aligned
+ */
+const gridFallbacks = css`
+  ${({ item, columns, parentColumns }) => {
+    if (!item || !parentColumns) {
+      return `position: relative;`;
+    }
+    const selectedGroups = Object.keys(columns);
+    return selectedGroups.map(
+      group =>
+        `
+      ${mediaQuery({
+        min: groups[group].min,
+        max: groups[group].max,
+        styles: `
+          display: inline-block;
+          width: ${(100 * columns[group]) / parentColumns[group]}%;
+          vertical-align: top; `,
+        /* [1] */
+      })}
+    `,
+    );
+  }}
+`;
+
+const GridComponent = styled.div`
+  ${({ enableGelMaxWidths }) => enableGelMaxWidths && gelMaxWidths}
+  ${({ enableGroupFourMaxWidth }) => enableGroupFourMaxWidth && group4MaxWidth}
+  ${gridFallbacks}
+  @supports (display: grid) {
+    ${gridMediaQueries}
+    ${({ item }) =>
+      item ? `display: block;` : `display: grid; position: initial;`}
+  }
+`;
+
+const Grid = ({ children, ...otherProps }) => {
+  const renderChildren = () =>
+    React.Children.map(children, child => {
+      return React.cloneElement(child, {
+        parentColumns: otherProps.columns,
+        parentStartOffset: otherProps.startOffset,
+      });
+    });
+
+  return <GridComponent {...otherProps}>{renderChildren()}</GridComponent>;
+};
+
+Grid.propTypes = {
+  children: node.isRequired,
+  columns: shape({
+    group1: number,
```

Comment:
IMO for now, all these groups should be required

Reply:
It seems the grid component handles not being passed groups:

> In this example groups 1, 2, 3, & 5 are omitted from the columns prop - so at those breakpoints items fill the whole width of the page.